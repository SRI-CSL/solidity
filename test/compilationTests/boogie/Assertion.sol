pragma solidity ^0.4.17;

contract Assertion {

    function abs_sound(int param) pure public returns (int) {
        int result = param;
        if (result < 0) result *= -1;
        assert(result >= 0);
        return result;
    }

    function abs_unsound(int param) pure public returns (int) {
        int result = param;
        if (result < 0) result += -1;
        assert(result >= 0);
        return result;
    }
}