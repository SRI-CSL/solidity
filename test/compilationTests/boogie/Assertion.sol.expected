Boogie program verifier finished with 2 verified, 1 error